eZ/Bundle/EzPublishCoreBundle/EventListener/RequestEventListener.php

```diff
@@ -71,10 +71,16 @@ public function onKernelRequestForward(GetResponseEvent $event)
                     $request->server->all(),
                     $request->getContent()
                 );
-                $forwardRequest->attributes->add($request->attributes->all());
-                if ($request->headers->has('X-User-Hash')) {
-                    $forwardRequest->headers->set('X-User-Hash', $request->headers->get('X-User-Hash'));
+
+                if ($request->attributes->has('forwardRequestHeaders')) {
+                    foreach ($request->attributes->get('forwardRequestHeaders') as $headerName => $headerValue) {
+                        $forwardRequest->headers->set($headerName, $headerValue);
+                    }
+                    $request->attributes->remove('forwardRequestHeaders');
```

Comment:
Note: There is a bug report from @Plopix  which I believe covers this logic regarding us setting vary by user hash on all respåomnses when it's set on incoming request _(which it is on Varnish)_.  While the feature is meant to be opt-in from controller side.

Reply:
yes this one: https://jira.ez.no/browse/EZP-28916